osu.Game.Rulesets.Mania/Mods/ManiaModFlashlight.cs

```diff
@@ -2,13 +2,56 @@
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
 using System;
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Mania.Objects;
 using osu.Game.Rulesets.Mods;
+using OpenTK;
 
 namespace osu.Game.Rulesets.Mania.Mods
 {
-    public class ManiaModFlashlight : ModFlashlight
+    public class ManiaModFlashlight : ModFlashlight<ManiaHitObject>
     {
         public override double ScoreMultiplier => 1;
         public override Type[] IncompatibleMods => new[] { typeof(ModHidden) };
+
+        private const float default_flashlight_size = 180;
+
+        public override Flashlight CreateFlashlight() => new ManiaFlashlight();
+
+        private class ManiaFlashlight : Flashlight
+        {
+            public ManiaFlashlight()
+            {
+                FlashlightSize = new Vector2(0, default_flashlight_size);
+            }
+
+            public override bool Invalidate(Invalidation invalidation = Invalidation.All, Drawable source = null, bool shallPropagate = true)
+            {
+                if ((invalidation & Invalidation.DrawSize) > 0)
+                {
+                    Schedule(() =>
+                    {
+                        FlashlightSize = new Vector2(DrawWidth, FlashlightSize.Y);
+
+                        FlashlightPosition = DrawPosition + DrawSize / 2;
```

Comment:
It's easier to set these values every frame in `Update()`, the base `FlashlightSize`/`FlashlightPosition` setters handle de-duping of values.